### PR TITLE
Slim 497 cucumber test monitoring

### PR DESCRIPTION
**Pull request by carlogeenen**
Implemented all monitoring features and requests. Created a new project with new directory and package structure to be prepared for more testcases in the future.

### CODE REVIEW

File: integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/support/TestCaseRunner.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+@Component
+public class TestCaseRunner {
+    public TestCaseResult runWsdlTestCase(final TestCase testCase, final String deviceId, final String organisationId,
+            final String correlationUid, final String testCaseNameRequest) throws XmlException, IOException,
+            SoapUIException {
+
+        final WsdlTestCase wsdlTestCase = (WsdlTestCase) testCase;
+        wsdlTestCase.setPropertyValue("DeviceIdentificationE", deviceId);
+        wsdlTestCase.setPropertyValue("OrganisationIdentification", organisationId);
+        wsdlTestCase.setPropertyValue("CorrelationUid", correlationUid);
```
Comment (hannek24) [inline integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/support/TestCaseRunner.java]: CorrelationId should not always be set. It should not be set for the request. Only for the request of the response.

File: integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/ActualMeterReads.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    @Autowired
+    private OrganisationId organisationId;
+
+    @When("^the get actual meter reads request is received$")
+    public void theGetActualMeterReadsRequestIsReceived() throws Throwable {
+        this.correlationUidPattern = Pattern.compile(this.organisationId.getOrganisationId()
+                + "\\|\\|\\|\\S{17}\\|\\|\\|\\S{17}");
+        this.testCase = this.wsdlProjectFactory.createWsdlTestCase(SOAP_PROJECT_XML, TEST_SUITE_XML, TEST_CASE_XML);
+        final TestCaseResult runTestStepByName = this.testCaseRunner.runWsdlTestCase(this.testCase,
+                this.deviceId.getDeviceId(), this.organisationId.getOrganisationId(), this.correlationUid,
```
Comment (hannek24) [inline integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/ActualMeterReads.java]: do not set the correlationId for the request. It is only applicable for the reqeust of the response.

File: integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/ActualMeterReadsGas.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    private OrganisationId organisationId;
+
+    @When("^the get actual meter reads gas request is received$")
+    public void theGetActualMeterReadsRequestIsReceived() throws Throwable {
+        this.correlationUidPattern = Pattern.compile(this.organisationId.getOrganisationId()
+                + "\\|\\|\\|\\S{17}\\|\\|\\|\\S{17}");
+        this.testCase = this.wsdlProjectFactory.createWsdlTestCase(SOAP_PROJECT_XML, TEST_SUITE_XML, TEST_CASE_XML);
+
+        final TestCaseResult runTestStepByName = this.testCaseRunner.runWsdlTestCase(this.testCase,
+                this.deviceId.getDeviceId(), this.organisationId.getOrganisationId(), this.correlationUid,
```
Comment (hannek24) [inline integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/ActualMeterReadsGas.java]: do not set the correlationId for the request. It is only applicable for the reqeust of the response.

File: integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/PeriodicMeterReads.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    private OrganisationId organisationId;
+
+    @When("^the get periodic meter reads request is received$")
+    public void theGetPeriodicMeterReadsRequestIsReceived() throws Throwable {
+        this.correlationUidPattern = Pattern.compile(this.organisationId.getOrganisationId()
+                + "\\|\\|\\|\\S{17}\\|\\|\\|\\S{17}");
+        this.testCase = this.wsdlProjectFactory.createWsdlTestCase(SOAP_PROJECT_XML, TEST_SUITE_XML, TEST_CASE_XML);
+
+        final TestCaseResult runTestStepByName = this.testCaseRunner.runWsdlTestCase(this.testCase,
+                this.deviceId.getDeviceId(), this.organisationId.getOrganisationId(), this.correlationUid,
```
Comment (hannek24) [inline integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/PeriodicMeterReads.java]: do not set the correlationId for the request. It is only applicable for the reqeust of the response.

File: integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/PeriodicMeterReadsGas.java [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+    private OrganisationId organisationId;
+
+    @When("^the get periodic meter reads gas request is received$")
+    public void theGetPeriodicMeterReadsRequestIsReceived() throws Throwable {
+        this.correlationUidPattern = Pattern.compile(this.organisationId.getOrganisationId()
+                + "\\|\\|\\|\\S{17}\\|\\|\\|\\S{17}");
+        this.testCase = this.wsdlProjectFactory.createWsdlTestCase(SOAP_PROJECT_XML, TEST_SUITE_XML, TEST_CASE_XML);
+
+        final TestCaseResult runTestStepByName = this.testCaseRunner.runWsdlTestCase(this.testCase,
+                this.deviceId.getDeviceId(), this.organisationId.getOrganisationId(), this.correlationUid,
```
Comment (hannek24) [inline integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/PeriodicMeterReadsGas.java]: do not set the correlationId for the request. It is only applicable for the reqeust of the response.

File: integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/Device.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package com.alliander.osgp.platform.cucumber.smartmeteringmonitoring;
```
Comment (hannek24) [inline integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/Device.java]: should not be in the smartmeteringmonitoring package, but in the com.alliander.osgp.platform.cucumber.smartmetering.device package for instance.

File: integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/Organisation.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package com.alliander.osgp.platform.cucumber.smartmeteringmonitoring;
```
Comment (hannek24) [inline integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/Organisation.java]: should not be in the smartmeteringmonitoring package, but in the com.alliander.osgp.platform.cucumber.smartmetering.organisation package for instance.

File: integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/ActualMeterReadsGas.java [hunk 8]
```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright 2016 Smart Society Services B.V. *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package com.alliander.osgp.platform.cucumber.smartmeteringmonitoring;
```
Comment (hannek24) [inline integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/ActualMeterReadsGas.java]: I would prefer to change the package name from 
com.alliander.osgp.platform.cucumber.smartmeteringmonitoring
do
com.alliander.osgp.platform.cucumber.smartmetering.monitoring

Comment (carlogeenen) [inline integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/ActualMeterReadsGas.java]: The current package naming is the same as the WSDL's they are referring to.

**Pull request merged**